### PR TITLE
gnunet-gtk: init at 0.12.0

### LINKED ISSUE #76754: package request: gnunet-gtk
**Project description**
It's a GUI for GNUnet written in GTK.
GNUnet is a new protocol stack for secure, distributed and privacy-preserving apps.

**Metadata**

* homepage URL: http://gnunet.org/
* source URL: https://git.gnunet.org/gnunet-gtk.git/
* license: gpl3+
* platforms: unix, linux, darwin, ...

### PR DESCRIPTION
**Pull request by pstn**
I worked on top of  #77785 for a newer gnunet version, so please only merge after that PR.

###### Motivation for this change
Fixes #76754 and I want it.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/p2p/gnunet-gtk/default.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  };
+
+  nativeBuildInputs= [ autoreconfHook wrapGAppsHook pkgconfig ];
+  buildInputs = [ libgcrypt libextractor libxml2 gnunet gnome3.glade gnutls gtk3 ];
+
+  patchPhase = "patchShebangs pixmaps/icon-theme-installer";
+
+  meta = with stdenv.lib; {
+    description = "GNUnet GTK User Interface";
+    homepage = https://github.com/ajf8/transmission-remote-gtk;
```
Comment (mehlon) [inline pkgs/applications/networking/p2p/gnunet-gtk/default.nix]: homepage = https://git.gnunet.org/gnunet-gtk.git

File: pkgs/applications/networking/p2p/gnunet-gtk/default.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    license = licenses.gpl3;
```
Comment (mehlon) [inline pkgs/applications/networking/p2p/gnunet-gtk/default.nix]: should be gpl3Plus

File: pkgs/applications/networking/p2p/gnunet-gtk/default.nix [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    maintainers = with maintainers;[ pstn ];
+    platforms = platforms.linux;
```
Comment (mehlon) [inline pkgs/applications/networking/p2p/gnunet-gtk/default.nix]: should be platforms.gnu ++ platforms.linux (same as gnunet)

**Comment by mehlon** [comment]
Ah, I realize I should have used the "suggest changes" option for GitHub instead. Oh well.

**Comment by pstn** [comment]
Rebased my branch on master, resolved the issues and rebuilt the package.

**Comment by pstn** [comment]
repushed just for signature

**Comment by pstn** [comment]
ping @jonringer for review before feature freeze.

File: pkgs/applications/networking/p2p/gnunet-gtk/default.nix [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  };
+
+  nativeBuildInputs= [ autoreconfHook wrapGAppsHook pkgconfig ];
+  buildInputs = [ libgcrypt libextractor libxml2 gnunet gnome3.glade gnutls gtk3 ];
+
+  patchPhase = "patchShebangs pixmaps/icon-theme-installer";
+
+  meta = with stdenv.lib; {
+    description = "GNUnet GTK User Interface";
+    homepage = https://git.gnunet.org/gnunet-gtk.git;
```
Comment (jonringer) [inline pkgs/applications/networking/p2p/gnunet-gtk/default.nix:23]: rfc 45
```suggestion
    homepage = "https://git.gnunet.org/gnunet-gtk.git";
```

Comment (pstn) [reply pkgs/applications/networking/p2p/gnunet-gtk/default.nix]: Good catch. Fixed.

**Review by jonringer — COMMENTED** [review]
installation seems to be missing some localization:
```
$ strace ./results//gnunet-gtk/bin/gnunet-fs-gtk
....
openat(AT_FDCWD, "/nix/store/jjcw4g52zjzw4nvqmrl2nfpqzvdlr58i-gnunet-gtk-0.12.0/share/gnunet//GNUnet GTK/locale//en_US.UTF-8/LC_MESSAGES/gnunet-gtk.mo", O_RDONLY) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/nix/store/jjcw4g52zjzw4nvqmrl2nfpqzvdlr58i-gnunet-gtk-0.12.0/share/gnunet//GNUnet GTK/locale//en_US.utf8/LC_MESSAGES/gnunet-gtk.mo", O_RDONLY) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/nix/store/jjcw4g52zjzw4nvqmrl2nfpqzvdlr58i-gnunet-gtk-0.12.0/share/gnunet//GNUnet GTK/locale//en_US/LC_MESSAGES/gnunet-gtk.mo", O_RDONLY) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/nix/store/jjcw4g52zjzw4nvqmrl2nfpqzvdlr58i-gnunet-gtk-0.12.0/share/gnunet//GNUnet GTK/locale//en.UTF-8/LC_MESSAGES/gnunet-gtk.mo", O_RDONLY) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/nix/store/jjcw4g52zjzw4nvqmrl2nfpqzvdlr58i-gnunet-gtk-0.12.0/share/gnunet//GNUnet GTK/locale//en.utf8/LC_MESSAGES/gnunet-gtk.mo", O_RDONLY) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/nix/store/jjcw4g52zjzw4nvqmrl2nfpqzvdlr58i-gnunet-gtk-0.12.0/share/gnunet//GNUnet GTK/locale//en/LC_MESSAGES/gnunet-gtk.mo", O_RDONLY) = -1 ENOENT (No such file or directory)
--- SIGSEGV {si_signo=SIGSEGV, si_code=SEGV_MAPERR, si_addr=0x48} ---
+++ killed by SIGSEGV (core dumped) +++
Segmentation fault (core dumped)
```

**Review by jonringer — APPROVED** [review]
it just seems to be that one command, and I can't think of a way to fix it.

**Comment by pstn** [comment]
I could reproduce the weird locale search path but not the segmentation fault. Curious.

**Pull request merged**